src/js/videojs-dash.js

```diff
@@ -70,6 +70,71 @@ class Html5DashJS {
     // element to bind to.
     this.mediaPlayer_.initialize();
 
+    // Retrigger a dash.js-specific error event as a generic HTML5 one
+    // See src/streaming/utils/ErrorHandler.js in dash.js code
+    // Handled with error (playback is stopped):
+    // - capabilityError
+    // - downloadError
+    // - manifestError
+    // - mediaSourceError
+    // - mediaKeySessionError
+    // Not handled:
+    // - timedTextError (video can still play)
+    // - mediaKeyMessageError (only fires under 'might not work' circumstances)
+    this.retriggerError_ = (event) => {
+      // No support for MSE
+      if ((event.error === 'capability' && event.event === 'mediasource') ||
+        // Manifest type not supported
+        (event.error === 'manifestError' && event.event.id === 'createParser') ||
+        // Codec(s) not supported
+        (event.error === 'manifestError' && event.event.id === 'codec') ||
+        // No streams available to stream
+        (event.error === 'manifestError' && event.event.id === 'nostreams') ||
+        // not sure about this one
+        (event.error === 'manifestError' && event.event.id === 'nostreamscomposed') ||
+        // syntax error parsing the manifest
+        (event.error === 'manifestError' && event.event.id === 'parse') ||
+        // streams are multiplexed, which is not supported by dash
+        (event.error === 'manifestError' && event.event.id === 'multiplexedrep')) {
+        this.player.error({code: 4});
```

Comment:
For these and the other errors, it might be useful to include a description of the errors as the message when triggering on the player. Having a clear error message may prevent reported issues when it's a problem with the content or browser (and just generally reduce confusion when an error occurs).

Reply:
Yeah, I was already wondering if I should at least pass through some of the more informative errors.